test/solidity/oz/ERC1967Upgrade/Vote.js

```diff
@@ -0,0 +1,188 @@
+        ['eip1967.proxy.implementation']
+      )
+
+      // uint256(keccak256('eip1967.proxy.implementation')) - 1
+      const eip1967ImplUint256 = ethers.BigNumber.from(eip1967ImplByte32).sub(1)
+
+      // bytes32(uint256(keccak256('eip1967.proxy.implementation')) - 1)
+      const expectedImplementationSlot = ethers.utils.hexZeroPad(
+        eip1967ImplUint256.toHexString(),
+        32
+      )
+
+      expect(await voteProxy.getImplementationSlot()).to.eq(
+        expectedImplementationSlot
+      )
+    })
+
+    it('Should deploy vote proxy contract with a new proxy admin', async () => {
+      expect(await voteProxy.getCurrentAdmin()).to.eq(await admin.getAddress())
+    })
+
+    it('Should be able to get the predefined ERC1967 ADMIN_SLOT', async () => {
+      // @logic ERC1967.ADMIN_SLOT is obtained as bytes32(uint256(keccak256('eip1967.proxy.admin')) - 1)
+
+      // keccak256('eip1967.proxy.admin')
+      const eip1967ImplByte32 = ethers.utils.solidityKeccak256(
+        ['string'],
+        ['eip1967.proxy.admin']
+      )
+
+      // uint256(keccak256('eip1967.proxy.admin')) - 1
+      const eip1967AdminUint256 =
+        ethers.BigNumber.from(eip1967ImplByte32).sub(1)
+
+      // bytes32(uint256(keccak256('eip1967.proxy.admin')) - 1)
+      const expectedAdminSlot = ethers.utils.hexZeroPad(
+        eip1967AdminUint256.toHexString(),
+        32
+      )
+
+      expect(await voteProxy.getAdminSlot()).to.eq(expectedAdminSlot)
+    })
+
+    it('Should be able to change the current proxy admin to a new address', async () => {
+      const tx = await voteProxy.changeAdmin(await voter.getAddress())
+      const receipt = await tx.wait()
+
+      const [previousAdmin, newAdmin] = receipt.events.map(
+        (e) => e.event === 'AdminChanged' && e
+      )[0].args
+
+      expect(previousAdmin).to.eq(await admin.getAddress())
+      expect(newAdmin).to.eq(await voter.getAddress())
+    })
+
+    it('Should NOT be able to change the current proxy admin if the caller is not an admin', async () => {
+      let error
+
+      const tx = await voteProxy
+        .connect(voter)
+        .changeAdmin(await voter.getAddress())
+
+      expect(tx.wait()).to.eventually.be.rejected.and.have.property(
+        'code',
+        CALL_EXCEPTION
+      )
+    })
+  })
+
+  describe('Implementation contract', () => {
+    beforeEach(async () => {
+      proxiedVoteV1 = new ethers.Contract(
+        voteProxy.address,
+        VoteV1Artifact.abi,
+        admin
+      )
+
+      await voteProxy.upgradeToAndCall(voteV2.address)
+
+      proxiedVoteV2 = new ethers.Contract(
+        voteProxy.address,
+        VoteV2Artifact.abi,
+        admin
+      )
+    })
+
+    it('Should load VoteV1 into proxy address', () => {
+      expect(proxiedVoteV1.address).to.eq(voteProxy.address)
+    })
+
+    it('Should cast votes to the system', async () => {
+      await proxiedVoteV1.connect(voter).vote()
+
+      const voters = await proxiedVoteV1.voters()
```

Comment:
Maybe expand this test or add another that confirms more than 1 voter is returned and the list is populated as expected